go.mod

```diff
@@ -0,0 +1,17 @@
+module sigs.k8s.io/sig-storage-lib-external-provisioner
```

Comment:
I believe the next release with this change will have to be v5.0.0 because introducing Go module support is a breaking change. Also, because the major version is > 1, versioned imports become mandatory.

This means that `sigs.k8s.io/sig-storage-lib-external-provisioner` has to be replaced with `sigs.k8s.io/sig-storage-lib-external-provisioner/v5` here and in all import statements in the code itself.

This also raises the question whether importing with dep is still supposed to work. It doesn't out-of-the-box anymore because dep doesn't understand versioned imports.

We worked around that with symlinks elsewhere, see https://github.com/kubernetes-csi/external-attacher/pull/209 and https://github.com/kubernetes-csi/csi-test/pull/232.

Reply:
a single commit is added to fix this.

Reply:
The commit message of that commit ("add compatibility with dep") misses the main reason for doing this: "go mod" will also refuse to pull the code unless the /v5 suffix matches the major version number.

Not sure whether you want to fix that.

Reply:
git message amended.

Reply:
I probably explained this wrong :grin: 

The "/v5" suffix isn't needed by dep, quite the opposite, it just gets confused by it. It's needed by "go mod". Anyway, let's leave the commit message as-is.

Reply:
Hm, I'm new to this pattern. I just read https://blog.golang.org/v2-go-modules and in their example they don't symlink, instead they keep v0/v1 in the top-level and they let all the versions coexist in the same branch.

I don't want to follow that example because this library makes breaking changes quite often and it would be a pain to retroactively make v2-v5 directories.

But I think symlinking could get pretty tedious as well. If I want to refactor and add/delete a file, I need to make sure the symlink under /v5 is updated accordingly? Is there a common script all our repos could share that could make validation & maintenance of these symlinks easier?

Reply:
What happens when I want to release a v6? Would I simply rename v5 to v6?

Reply:
> I don't want to follow that example because this library makes breaking changes quite often and it would be a pain to retroactively make v2-v5 directories.

I agree, having multiple different versions in the same branch looks awkward. It's probably only relevant if developers really need to support multiple API versions.

Symlinking is a stop-gap measure to continue supporting dep. I don't know how important that is for the consumers of this library. It is something that could be scripted, but there is no script at the moment.

> What happens when I want to release a v6? Would I simply rename v5 to v6?

Yes.